### ISSUE TITLE
clarify board version in constraint file or name?

### ISSUE THREAD

**Issue by gojimmypi** [emard/ulx3s-passthru#2]
the constraint file listed in: [ulx3s/doc/constraints/ulx3s_v20.lpf](https://github.com/emard/ulx3s/blob/master/doc/constraints/ulx3s_v20.lpf) notes "_ULX3S v2.x.x and v3.0.x_" - however the one listed  here in  [ulx3s-passthru/constraints/ulx3s_v20.lpf](https://github.com/emard/ulx3s-passthru/blob/master/constraints/ulx3s_v20.lpf) notes only "_ULX3S v2.0 and v2.1_".  Although the same name, the files contents are quite different.

is this local constraint file in `ulx3s-passthru` really for only the ULX3S v2.0 and v2.1 board? 

perhaps it would be good to identify a master source of constraint files and explicitly state their version and in the text of each file? maybe also give each of the boards their own named constraint file?

**Comment by emard** [comment]
No it's for all v20 and later boards.
At the time passthru was written, there was no
v3.0.3 boards

Every project keeps its old copy of constraints
because it has beend thoroughly TESTED

changing just constraints for cosmetic reasons
can potentially break function :)

**Comment by gojimmypi** [comment]
yes, I see the value in having a single constraints file per project. thanks for the clarification that it is for v20 and later boards; (perhaps say this in the file?)

However my https://github.com/emard/ulx3s/pull/3 was not for cosmetic reasons. Rather, once Diamond has written the file out, doing a compare / diff to see new hand-assigned changes is almost impossible - as it appears practically the entire file has changed (and the ignore whitespace does not help). This PR in a non-project, master location was more to allow easier tracking of changes - say when that file gets updated and I want to use it: what's different in the local Diamond file vs the new file.

just a suggestion/idea :)  I do see the concern of breaking existing things. Do you have any examples of whitespace breaking a project or otherwise causing undesired results?

**Comment by emard** [comment]
Some projects are "essential" to be unbroken like
passthru, it's simple as blinkled but must work :)

Probably not for spaces but sometimes a complex and
overclocked project like amiga fails if
constraints are changed in naive way e.g.
gpdi0 gpdi1 gpdi2 is rewritten as vector gpdi[2:0].

For fresh new examples, you can take constraints,
modify and reformat what you wish, remove whitespace.

You can decide is it cleaner/better so that all examples
take constraints from the same file or each example to have
its own
